contrib/linux/actions/wait_for_ssh.py

```diff
@@ -2,26 +2,36 @@
 
 import time
 
-import paramiko
+from oslo_config import cfg
 
 from st2actions.runners.pythonrunner import Action
+from st2actions.runners.ssh.paramiko_ssh import ParamikoSSHClient
 
 
 class BaseAction(Action):
-    def run(self, keyfile, username, hostname, ssh_timeout, retries):
-        key = paramiko.RSAKey.from_private_key_file(keyfile)
-        client = paramiko.SSHClient()
-        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    def run(self, hostname, port, username, password=None, keyfile=None, ssh_timeout=5,
+            sleep_delay=20, retries=10):
+        # Note: If neither password nor key file is provided, we try to use system user
+        # key file
+        if not password and not keyfile:
```

Comment:
hmm, not really our concern but I doubt we will be dinged for being opinionated.